src/Raven.Server/RavenServer.cs

```diff
@@ -203,49 +203,27 @@ public void Initialize()
                     {
                         try
                         {
-                            var msg = reader.ReadLine();
-                            var tokens = msg.Split(new[] {' '}, StringSplitOptions.RemoveEmptyEntries);
-
-                            if (tokens.Length < 1)
+                            var cli = new RavenCli();
+                            var restart = cli.Start(this, writer, reader, false);
+                            if (restart)
                             {
-                                var reply = "Unknown command";
-                                PipeLogAndReply(writer, reply);
-                                continue;
+                                writer.WriteLine("Restarting Server...<DELIMETER_RESTART>");
+                                // TODO :: ADIADI :: perform restart server
```

Comment:
here we identify the remote cli "wants" quit or restart.  The problem is we cannot perform a restart to the Console.ReadLine thread. However we can change few things in order to achieve that differently so it is under "TODO" (if you type in remote connection "q" - you reach here).